### PR TITLE
Fix linkerd app

### PR DESCRIPTION
**Pull request by alexeigutium**
## BACKGROUND
* Linkerd deploy file is outdated and the app can't be installed

-----------------------------------------------------------------------

## Changes
* Update linkerd version from 2.11.2 to 2.12.4
* Add required linkerd crds installation
* Add dirty fix for failing kubectl rollout command

-----------------------------------------------------------------------

## Checklist
- [X] review the [contributing doc](https://github.com/digitalocean/marketplace-kubernetes/blob/master/CONTRIBUTING.md) with steps for both adding or updating your application (if applicable)
------------------------------------------------------------------------

Reviewer: @marketplace-eng

### CODE REVIEW

File: stacks/linkerd2/upgrade.sh [hunk 1]
```diff
@@ -33,7 +33,10 @@ $BINARY upgrade | kubectl apply --prune -l linkerd.io/control-plane-ns=linkerd \
 # ensure services are running
 kubectl get deployments -o custom-columns=NAME:.metadata.name | tail -n +2 | while read -r line
 do
-  kubectl rollout status -w deployment/"$line"
+  if [ ! -z "$line" ]
+  then 
+    kubectl rollout status -w deployment/"$line"
+  fi
```
Comment (v-aisac) [inline stacks/linkerd2/upgrade.sh:39]: Maybe we need an else statement with an error drop?

File: stacks/linkerd2/deploy.sh [hunk 2]
```diff
@@ -23,12 +23,20 @@ wget -q $URL -O "$BINARY" && chmod +x "$BINARY"
 kubectl config set-context --current --namespace=linkerd
 
 # deploy linkerd
+$BINARY install --crds --ignore-cluster | kubectl apply -f -
 $BINARY install --ignore-cluster | kubectl apply -f -
 
 # ensure services are running
+kubectl get deployments -o custom-columns=NAME:.metadata.name
+
 kubectl get deployments -o custom-columns=NAME:.metadata.name | tail -n +2 | while read -r line
```
Comment (mbeyt) [inline stacks/linkerd2/deploy.sh:32]: Is there a reason you're running this command twice? Same with the install command on line 26 and 27.

Comment (alexeigutium) [reply stacks/linkerd2/deploy.sh:32]: no, I used for debugging and forgot to remove

File: stacks/linkerd2/upgrade.sh (hunk 1 above)
Comment (alexeigutium) [reply stacks/linkerd2/upgrade.sh:39]: hmmmm

Comment (alexeigutium) [reply stacks/linkerd2/upgrade.sh:41]: thank you, added, but I couldn't test it. 🤞🏿

**Review by mbeyt — APPROVED** [review]

**Pull request merged**